extensions/azurecore/src/utils/Logger.ts

```diff
@@ -6,19 +6,52 @@
 import * as vscode from 'vscode';
 import * as loc from '../localizedConstants';
 
+export enum LogLevel {
+	'All',
+	'Off',
+	'Critical',
+	'Error',
+	'Warning',
+	'Information',
+	'Verbose',
+}
+
 export class Logger {
 
-	public static channel: vscode.OutputChannel = vscode.window.createOutputChannel(loc.extensionName);
 
+	public static channel: vscode.OutputChannel = vscode.window.createOutputChannel(loc.extensionName);
 	private static _piiLogging: boolean = false;
+	public static config = vscode.workspace.getConfiguration('mssql');
 
-	static log(msg: any, ...vals: any[]) {
-		const fullMessage = `${msg} - ${vals.map(v => JSON.stringify(v)).join(' - ')}`;
-		this.channel.appendLine(fullMessage);
+	static write(logLevel: LogLevel, msg: any, ...vals: any[]) {
```

Comment:
It'd be nice to have helper functions `Logger.error`, `Logger.information` etc that automatically add the appropriate log level.

Reply:
What do you mean by this? I specify the LogLevel when I call the write function

Reply:
I'm saying that it'd be simpler to just have separate error, information, etc. functions that then call the current Logger.write function with the appropriate log level. Just makes it easier to read and use

Reply:
(which we already have with Logger.error anyways, so you might as well stay consistent as well)